src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1372,4 +1374,21 @@ public static void sendRecipeBook(NetHandlerPlayServer connection, State state,
         if (recipes.size() > 0 || display.size() > 0)
             connection.sendPacket(new SPacketRecipeBook(state, recipes, display, isGuiOpen, isFilteringCraftable));
     }
+    
+    /**
+     * Return if the passed player currently has the ability to use elytra flight
+     * Return true if the stack is vanilla elytra for non player EntityLivingBase to emulate exact vanilla behaviour 
+     * @param entity
+     * @return
+     */
+    public static boolean hasElytraFlightAbility(EntityLivingBase player, ItemStack itemstack)
+    {
+    	boolean vanillaElytra = itemstack.getItem() == Items.ELYTRA && ItemElytra.isUsable(itemstack);
+    	if (!(player instanceof EntityPlayer))
```

Comment:
What is the purpose of this `EntityPlayer` check? I don't think you mentioned it anywhere.

Reply:
Because this attribute only applies to players.

Reply:
If it is only for players, should the attribute be a field next to `EntityPlayer.REACH_DISTANCE`, and not in `SharedMonsterAttributes`?

Reply:
The check is because this method is called from EntityLivingBase. And the Entity Attribute only applies to players. The way the method is coded it behaves identically to vanilla code when called from a non-player ELB.

Yes the Entity Attribute could be added to player instead of SharedMonsterAttributes. When I first made this PR the reach distance was not merged yet, now It definitely makes sense to put it next to the REACH_DISTANCE to reduce patches. I will move this and update the PR today.

Reply:
Or it can be just made into an attribute for all living entities.

Reply:
But is this needed? I don't think it makes much sense, since the flight code is for players. Entity movement is controlled by AI.

Reply:
But is there any reason why an AI can't take advantage of an entity that has elytra?

Reply:
There isn't, but you can add this attribute to an entity if you want to, and if you need an elytra flight on your npc, you would have your own code anyway. The point of this PR is to un-hardcode the elytra check for players, because currently the only way to give players elytra flight is with core modding. And an entity attribute is the most compatible way when multiple mods want to add elytra flight. Throwing a CanPlayerFlyElytraEvent every tick doesn't sound like a good idea.

I personally do not care where the static Entity Attribute is stored. I just want the PR merged ;-). Adding it to EntityPlayer would reduce patches. And if a custom entity really wants to use it, you can still access the static final field from anywhere.